phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -2285,6 +2317,7 @@ public void createTable(RpcController controller, CreateTableRequest request,
                 builder.setReturnCode(MetaDataProtos.MutationCode.TABLE_NOT_FOUND);
                 if (indexId != null) {
                     builder.setViewIndexId(indexId);
```

Comment:
In case the create table request was made from an older client you need to set the viewIndexId as an int32 (as thats what the client is expecting). See my other comment about maintaing backward compatibility.